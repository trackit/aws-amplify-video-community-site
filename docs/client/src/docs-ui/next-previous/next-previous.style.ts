import {css} from "emotion";

export const nextPreviousLinkContainerStyle = css`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 1rem 0;
`;

export const nextPreviousLinkStyle = css`
  &:first-child {
    margin-right: 1rem;
  }

  a {
    display: flex;
    flex-direction: row;

    &:hover {
      text-decoration: none;
    }
  }

  h4,
  span {
    color: var(--font-color);
    text-align: right;
  }

  span {
    text-transform: uppercase;
    font-size: 0.75rem;
  }

  img {
    margin-left: 1rem;
    width: 0.5rem;
  }

  &:first-child {
    img {
      margin-left: 0;
      margin-right: 1rem;
    }

    h4,
    span {
      text-align: left;
    }
  }
`;

export const textGroupStyle = css`
  display: flex;
  flex-direction: column;

  > * {
    display: block;
  }
`;

export const textAlignmentStyleByDirection = {
  next: css`
    align-items: flex-end;
  `,
  previous: css`
    align-items: flex-start;
  `,
};
