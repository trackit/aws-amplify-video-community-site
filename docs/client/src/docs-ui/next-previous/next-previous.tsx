import {Component, h, Host, Prop} from "@stencil/core";
import {
  nextPreviousLinkContainerStyle,
  nextPreviousLinkStyle,
  textGroupStyle,
  textAlignmentStyleByDirection,
} from "./next-previous.style";
import {Page, PageLink, Direction} from "../../api";
import {pageContext} from "../page/page.context";
import {SelectedFilters} from "../page/page.types";
import {VNode} from "@stencil/router/dist/types/stencil.core";

@Component({tag: "docs-next-previous", shadow: false})
export class DocsNextPrevious {
  /*** page from with `next` and `previous` can be attained */
  @Prop() readonly page?: Page;
  /*** currently-selected filter state */
  @Prop() readonly selectedFilters?: SelectedFilters;

  renderLink = (pageLink: PageLink, direction: Direction): VNode => {
    return (
      <docs-internal-link class={nextPreviousLinkStyle} href={pageLink.route}>
        {direction === "previous" && (
          <img src="/assets/arrow-left.svg" alt="Previous Page" />
        )}
        <div
          class={{
            [textGroupStyle]: true,
            [textAlignmentStyleByDirection[direction]]: true,
          }}
        >
          <span>{direction}</span>
          <h4>{pageLink.title}</h4>
        </div>
        {direction === "next" && (
          <img src="/assets/arrow-right.svg" alt="Next Page" />
        )}
      </docs-internal-link>
    );
  };

  render() {
    return (
      <Host class={nextPreviousLinkContainerStyle}>
        {this.page &&
          [
            {
              direction: "previous" as Direction,
              pageLinkOrLinkDictionary: this.page.previous,
            },
            {
              direction: "next" as Direction,
              pageLinkOrLinkDictionary: this.page.next,
            },
          ].map(({direction, pageLinkOrLinkDictionary}) => {
            if (pageLinkOrLinkDictionary) {
              if (pageLinkOrLinkDictionary.route) {
                return this.renderLink(
                  pageLinkOrLinkDictionary as PageLink,
                  direction,
                );
              } else {
                const filterKey = this.page?.filterKey;
                if (filterKey && this.selectedFilters) {
                  const selected = this.selectedFilters[filterKey] as string;
                  const currentPageLink = pageLinkOrLinkDictionary[
                    selected
                  ] as PageLink;
                  if (currentPageLink) {
                    return this.renderLink(currentPageLink, direction);
                  }
                }
              }
            }

            return <div />;
          })}
      </Host>
    );
  }
}

pageContext.injectProps(DocsNextPrevious, ["selectedFilters"]);
