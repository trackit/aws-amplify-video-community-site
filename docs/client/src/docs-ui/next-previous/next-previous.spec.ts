import {DocsNextPrevious} from "./next-previous";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-next-previous", () => {
  describe("Component logic", () => {
    let nextPrevious: DocsNextPrevious;
    beforeEach(() => (nextPrevious = new DocsNextPrevious()));

    it("should init `page` as `undefined`", () =>
      expect(nextPrevious.page).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsNextPrevious],
            html: `<docs-next-previous />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
