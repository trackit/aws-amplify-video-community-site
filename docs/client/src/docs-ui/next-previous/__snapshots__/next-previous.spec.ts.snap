// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-next-previous Render logic should render 1`] = `<docs-next-previous class="css-20cstw"></docs-next-previous>`;
